src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -204,9 +204,29 @@ String decrypt(final String encryptedCredential) {
      * @param template the template
      * @return the redacted template
      */
-    public Template redactTemplateCredentials(Template template) {
+    public Template redactTemplateSecuredFields(Template template) {
+        Template updatedTemplate = null;
+
+        if (template.getUser() != null) {
+            updatedTemplate = new Template.Builder(template).name(template.name())
+                .description(template.description())
+                .useCase(template.useCase())
+                .templateVersion(template.templateVersion())
+                .user(null)
+                .uiMetadata(template.getUiMetadata())
+                .compatibilityVersion(template.compatibilityVersion())
+                .workflows(template.workflows())
+                .createdTime(template.createdTime())
+                .lastUpdatedTime(template.lastUpdatedTime())
+                .lastProvisionedTime(template.lastProvisionedTime())
+                .build();
+        } else {
+            updatedTemplate = template;
+        }
+
         Map<String, Workflow> processedWorkflows = new HashMap<>();
-        for (Map.Entry<String, Workflow> entry : template.workflows().entrySet()) {
+
+        for (Map.Entry<String, Workflow> entry : updatedTemplate.workflows().entrySet()) {
```

Comment:
These are the same workflows as before.  Seems we could move this workflow redaction step above the user check and replace both fields (user always with null and new workflows).

Reply:
Not sure if I understand this correctly but I moved the user check to later. Let me know if you meant something else and I will make the changes

Reply:
What I meant to say was that `template.workflows()` and `updatedTemplate.workflows()` are exactly the same, as the update only (possibly) changed user, so this particular variable change doesn't really mean anything (although it's more technically correct).

There are two steps: (1) conditionally replace user with null if it's non-null, and (2) always replace workflows.  Those steps could be combined into one step replacing user (always null even if it was before) and workflows.